### ISSUE TITLE
Fix: a11y - Lack of association for left label and checkbox 544

### ISSUE THREAD

**Issue by jpante** [Workday/canvas-kit#363]
## Bug Report

 Label is not announced in SR announcement. Reads "Checkbox option
check box not checked".

## Expected Behavior

 Label is announced. Reads "Label Checkbox option
check box not checked"
